pkg/segment/test/segment_test.go

```diff
@@ -98,6 +116,62 @@ func TestAddRandom(t *testing.T) {
 			t.Errorf("Iteration %d: %v", i, err)
 			break
 		}
+		if err := s.checkMaxGap(); err != nil {
```

Comment:
Per above, the fact that this is called from test code only means it should probably be a method in the test file itself. Just make it like this:

func checkMaxGap(n *node) error {
   ...
}

Reply:
As above, this actually can't be implemented outside as it requires access to private member hasChildren, nrSegments, maxGap and so on.